includes/wcpdf-functions.php

```diff
@@ -721,3 +721,190 @@ function wpo_wcpdf_get_multilingual_languages(): array {
+		}
+	}
+
+	// Fallback to 'finfo_file' if mime type is empty for local files only (no remote files allowed)
+	if ( empty( $mime_type ) && function_exists( 'finfo_open' ) && ! filter_var( $src, FILTER_VALIDATE_URL ) ) {
+		$finfo = finfo_open( FILEINFO_MIME_TYPE );
+
+		if ( $finfo ) {
+			$mime_type = finfo_file( $finfo, $src );
+			finfo_close( $finfo );
+		} else {
+			wcpdf_log_error( 'Fileinfo failed to open for file: ' . $src );
+		}
+	}
+
+	// Handle remote files
+	if ( empty( $mime_type ) && filter_var( $src, FILTER_VALIDATE_URL ) ) {
+		$context = stream_context_create( array(
+			'http' => array(
+				'method'        => 'HEAD',
+				'ignore_errors' => true,
+			),
+			'https' => array(
+				'method'           => 'HEAD',
+				'ignore_errors'    => true,
+				'verify_peer'      => false,
+				'verify_peer_name' => false,
+			),
+		) );
+
+		$headers = @get_headers( $src, 1, $context );
+
+		if ( $headers ) {
+			if ( isset( $headers['Content-Type'] ) ) {
+				$mime_type = is_array( $headers['Content-Type'] ) ? $headers['Content-Type'][0] : $headers['Content-Type'];
+			} else {
+				wcpdf_log_error( 'Content-Type header not found for URL: ' . $src );
+			}
+		} else {
+			wcpdf_log_error( 'Failed to get headers for URL: ' . $src );
+		}
+	}
+
+	// Determine using WP functions
+	if ( empty( $mime_type ) ) {
+		$path      = wp_parse_url( $src, PHP_URL_PATH );
+		$file_info = wp_check_filetype( $path );
+		$mime_type = $file_info['type'] ?? 'application/octet-stream';
+
+		if ( 'application/octet-stream' === $mime_type ) {
+			wcpdf_log_error( 'Unknown file extension for file: ' . $src );
+		}
+	}
+
+	// Last chance, determine from file extension
+	if ( empty( $mime_type ) ) {
+		$path      = parse_url( $src, PHP_URL_PATH );
+		$extension = strtolower( pathinfo( $path, PATHINFO_EXTENSION ) );
+
+		switch ( $extension ) {
+			case 'jpg':
+			case 'jpeg':
+				$mime_type = 'image/jpeg';
+				break;
+			case 'png':
+				$mime_type = 'image/png';
+				break;
+			case 'gif':
+				$mime_type = 'image/gif';
+				break;
+			case 'bmp':
+				$mime_type = 'image/bmp';
+				break;
+			case 'webp':
+				$mime_type = 'image/webp';
+				break;
+			case 'svg':
+				$mime_type = 'image/svg+xml';
+				break;
+			default:
+				$mime_type = 'application/octet-stream';
+				wcpdf_log_error( 'Unknown file extension: ' . $extension . ' for file: ' . $src );
+		}
+	}
+
+	return $mime_type;
+}
+
+/**
+ * Base64 encode file from URL or local path
+ *
+ * @param string $src
+ *
+ * @return string|bool
+ */
+function wpo_wcpdf_base64_encode_file( string $src ) {
+	$file_data = @file_get_contents( $src );
+	return base64_encode( $file_data ) ?? false;
+}
+
+/**
+ * Check if a file is readable
+ *
+ * @param string $path
+ * @return bool
+ */
+function wpo_wcpdf_is_file_readable( string $path ): bool {
+	// Check if the path is a URL
+	if ( filter_var( $path, FILTER_VALIDATE_URL ) ) {
+		$response = wp_safe_remote_head( $path );
```

Comment:
When I view the PDF, the path is like this:
`C:/laragon/www/mohamad/wp-content/uploads/2024/04/DALL.png`

When I view the HTML, the path is like this:
`https://localhost/mohamad/wp-content/uploads/2024/04/DALL.png`

In the HTML, the logo is not displayed for me, as I get this response:
`{"errors":{"http_request_failed":["cURL error 60: SSL certificate problem: self signed certificate"]},"error_data":[]}`